probe-rs/src/flashing/flash_algorithm.rs

```diff
@@ -208,38 +208,47 @@ impl FlashAlgorithm {
         let mut addr_stack = 0;
         let mut addr_load = 0;
         let mut addr_data = 0;
+        let mut code_start = 0;
 
         // Try to find a stack size that fits with at least one page of data.
         for i in 0..Self::FLASH_ALGO_STACK_SIZE / Self::FLASH_ALGO_STACK_DECREMENT {
-            offset = Self::FLASH_ALGO_STACK_SIZE - Self::FLASH_ALGO_STACK_DECREMENT * i;
-            // Stack address
-            addr_stack = ram_region.range.start + offset;
             // Load address
-            addr_load = addr_stack;
+            addr_load = raw
+                .load_address
+                .map(|a| a - (header.len() * size_of::<u32>()) as u32) // adjust the raw load address to account for the algo header
```

Comment:
We have to check for an underflow here, `raw.load_address` could be zero.

Reply:
Good spot! Fixed :)